src/Libraries/RevitNodes/Elements/Floor.cs

```diff
@@ -161,6 +165,81 @@ public static Floor ByOutlineTypeAndLevel(PolyCurve outline, FloorType floorType
             return new Floor(ca, floorType.InternalFloorType, level.InternalLevel );
         }
 
+
+        #endregion
+
+        #region Public Properties
+
+        /// <summary>
+        /// Get Slab Shape Points
+        /// </summary>
+        public IEnumerable<Pt> Points
+        {
+            get 
+            {
+                if (this.InternalFloor.SlabShapeEditor == null)
+                {
+                    throw new Exception(Properties.Resources.InvalidShapeEditor);
+                }
+
+                TransactionManager.Instance.EnsureInTransaction(DocumentManager.Instance.CurrentDBDocument);
+                this.InternalFloor.SlabShapeEditor.Enable();
+                TransactionManager.Instance.TransactionTaskDone();
+
+                List<Pt> points = new List<Pt>();              
+                foreach (SlabShapeVertex v in this.InternalFloor.SlabShapeEditor.SlabShapeVertices)
+                {
+                    points.Add(v.Position.ToPoint());
+                }
+                return points;
+            }
+        }
+
+        /// <summary>
+        /// Add Point to Slab Shape
+        /// </summary>
+        public void AddPoint(Pt point)
```

Comment:
@Randy-Ma but this would act differently since the slab shape editor affects the original element. The floor you used as an input is modified afterwards. I could still return it though.